Source/driver/Castro_advance.cpp

```diff
@@ -425,9 +425,7 @@ Castro::finalize_do_advance(Real time, Real dt, int amr_iteration, int amr_ncycl
 
 	dSdt_new.setVal(0.0, NUM_GROW);
 
-	for (int n = 0; n < num_src; ++n) {
-	    MultiFab::Add(dSdt_new, *new_sources[n], Xmom, Xmom, 3, 0);
-	}
+        MultiFab::Add(dSdt_new, new_sources, Xmom, Xmom, 3, 0);
```

Comment:
why do we only need dSdt for the momenta? and not other quantities?

Reply:
The comment above that line addresses this:

"Since we want (S^{n+1} - S^{n}) / dt, we only need to take twice the new-time source term, since in the predictor-corrector approach, the new-time source term is 1/2 * S^{n+1} - 1/2 * S^{n}. This is untrue in general for the non-momentum sources, but those don't appear in the hydro anyway, and for safety we'll only do this on the momentum terms."

I'm not sure what is meant by this not being the case for non-momentum sources, as that was quite a while ago.